File: pyworks/classfication/person.py
#Person 클래스 정의
class Person:
    def __init__(self, name, age, add):          #생성자 함수
        self.name = name
        self.age = age
        self.add=add
    #맴버의 정보
    def __str__(self):
        return f'이름:{self.name}, 나이:{self.age}, 주소:{self.add}'

#상속 -자식 클래스 이름 (부모클래스)
class Employee(Person):
    def __init__(self, name, age, add, id):
        super().__init__(name, age, add)
        self.id = id

        #매서드 재정의(오버라이드)
    def __str__(self):
        #return f'이름:{self.name}, 나이:{self.age}, 주소:{self.add}, 사번: {self.id}'
        return f'{super().__str__()}, 사번: {self.id}'
e1 = Employee("이하나", 25, "서울", 'a1001')
print(e1)

"""
#클래스 사용- 객체 생성(메모리 실행)
# p1 -인스턴스 , 오브젝트

p1 = Person("김산", 21, "강남")
print(p1.name)
print(p1.age)
print(p1)

p2= Person("이강", 30, "인천")
print(p2.name)
print(p2.age)
print(p2)
"""

File: javaworks/DBConnection/src/db/Person.java
package db;
//vo(value object
public class Person {

		private String userId;
		private String userpw;
		private String name;
		private int age;
		//기본생성자
		public Person() {}
		
		//외부 입력 가능한 생성자
		public Person(String userId, String userpw,
						String name, int age) {
			this.userId = userId;
			this.userpw = userpw;
			this.name = name;
			this.age =age;

		}
		
		//setter,getter
		public String getUserId() {
			return userId;
		}
		public void setUserId(String userId) {
			this.userId = userId;
		}
		public String getUserpw() {
			return userpw;
		}
		public void setUserpw(String userpw) {
			this.userpw = userpw;
		}
		public String getName() {
			return name;
		}
		public void setName(String name) {
			this.name = name;
		}
		public int getAge() {
			return age;
		}
		public void setAge(int age) {
			this.age = age;
		}
		
	
}
